extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -362,4 +373,47 @@ private synchronized void setPatchMatchingPolicyEnabled() {
 
         protected abstract boolean httpPermissionsEmpty();
     }
+
+    public void setMtlsCertificateRoleProperties(HttpConfiguration config) {
+        if (config.auth.certificateRoleProperties.isPresent()) {
+            Map<String, Set<String>> roles = new HashMap<>();
+
+            Path rolesPath = config.auth.certificateRoleProperties.get();
+            URL rolesResource = null;
+            if (Files.exists(rolesPath)) {
+                try {
+                    rolesResource = rolesPath.toUri().toURL();
+                } catch (MalformedURLException e) {
+                    throw new ConfigurationException(
+                            "quarkus.http.auth.mtls-certificate-role-properties is malformed",
+                            Set.of("quarkus.http.auth.mtls-certificate-role-properties"));
+                }
+            } else {
+                rolesResource = Thread.currentThread().getContextClassLoader().getResource(rolesPath.toString());
+            }
+            Properties rolesProps = new Properties();
+            try (InputStream is = rolesResource.openStream()) {
+                rolesProps.load(is);
+                for (Map.Entry<Object, Object> e : rolesProps.entrySet()) {
+                    log.debugf("Added role mapping for %s:%s", e.getKey(), e.getValue());
+                    roles.put((String) e.getKey(), parseRoles((String) e.getValue()));
+                }
+            } catch (Exception e) {
+                log.warnf("Unable to read roles mappings from %s:%s", rolesPath, e.getMessage());
+            }
+            InstanceHandle<MtlsAuthenticationMechanism> mtls = Arc.container().instance(MtlsAuthenticationMechanism.class);
+            if (mtls.isAvailable()) {
```

Comment:
this is **fine**, let's keep it, but tbh I tried to find when this could happen and I can't think of any situation unless there is bug in Quarkus